assets/scripts/app.js

```diff
@@ -75,8 +75,28 @@ $(() => {
       autoGenerateState = !autoGenerateState
     }
   })
-  $('#icon-palette').on('click', (event) => {
-    event.preventDefault()
-    colorGenerator.makeColors()
-  })
+  // $('#icon-palette').on('click', (event) => {
+  //   event.preventDefault()
+  //   colorGenerator.makeColors()
+  // })
+
+ // COLOR CODE COPY TO CLIPBOARD
+ const colorCodes = document.querySelectorAll('.color-codes')
+ colorCodes.forEach(div => div.addEventListener('click', function(){
+  let hexCodes = div.childNodes[1].value
+  const el = document.createElement('textarea')
+  el.value = hexCodes
+  document.body.appendChild(el)
+  el.select()
+  document.execCommand('copy')
+  document.body.removeChild(el)
+  $('#message').text('Color copied to the clipboard!').attr('class', 'alert alert-success shadow-lg')
+  $('#message').fadeIn()
+  setTimeout(() => { $('#message').fadeOut() }, 2000)
+  $('#saveButton').attr('disabled', true)
+ }))
+
```

Comment:
Event listener tehlikeli bir fonksiyon, ozellikle de `forEach` gibi bir iterator kullaniyorsan. Bu sekilde ilerleyerek yazdigin bir uygulamanin ilerde cidde performans sikintilari olabilir. Ayrica, ne kadar cok kod = o kadar yavas kod. Bu yazdigini `div.color-codes` elementlerini `button` yaparak uygularsan daha kisa bir kod elde edebilirsin.
Ornegin:
```html
<button type="button" class="color-codes">
    <input />
</button>
```

```javascript
const copyToClipboard = element => {
  element.select()
  document.execCommand('copy')
  return element.value
}

document.querySelector('button.color-codes').onclick = (e) => {
  const value = copyToClipboard(e.target.childNodes[1])
  alert(`${value} is copied on clipboard!`)
}
```
Ya da:
```javascript
$('button.color-codes').on('click', e => {
  const val = copyToClipboard(e.target.childNodes[1])
  alert(`${val} is copied on clipboard!`)
})
```
Tabi button elementi belli bir style ile geliyor, css ile onlari sifirlaman lazim.

Reply:
Bunun calismasi icin `<input />` elementlerindeki `disable` attribute 'unu kaldirman lazim. 
O `disable` sana anlattigim sey yuzunden var, direkt hex degerlerini degistirebiliyorsun. Bazi sikintilari vardi, mesela renk acik renkli olunca yazilar beyaz oldugu icin adam akilli gorunmuyordu falan, `disable` yaptim vaktim olmadigi icin. Su ise yariyor bak:
https://drive.google.com/file/d/14HTO8nfhXDUw5vyIKNgr1OIeDEDPpGLI/view?usp=sharing

Reply:
Disable attribute'unu kaldırıp denedim dediğin şekilde, ama hex code bilmeyen biri için bence bir anlam ifade etmiyor. Aslında önce code üzerine tıkladığın zaman copy yapması için uğraştım ama bir türlü istediğim gibi seçmedi kodu. Onun yerine parent'ını kullanmak zorunda kaldım.